### PR TITLE
Ch setup restapi project

### PR DESCRIPTION
**Pull request by Isaackomeza**
#### What does this PR do?

- Setting up project structure
- Setting up project welcoming endpoint
- Configuring ESL-int for code analysis

#### Description of Task to be completed?

- Initial setup of project structure
- Installing node_modules and required packages

#### How should this be manually tested?

- Use **http://localhost:3000** in postman to test endpoint
- Use `npm run lint` to test ESlint

#### Any background context you want to provide?

- With ESlint we can track and analyse our code

#### What are the relevant trello stories?

- https://trello.com/b/bt2IQfEv/back-end-phantom

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/Q

### CODE REVIEW

**Review by Rwandarushya — APPROVED** [review]
Well Done

**Review by malaba6 — CHANGES_REQUESTED** [review]
Good job Isaac. Looks good here. One thing thought, please make sure to squash those commits into one. I  will merge this only if it has one commit.

**Review by joshwambere — APPROVED** [review]
good job

**Review by joshwambere — COMMENTED** [review]
good job

**Comment by Isaackomeza** [comment]
Hey, I'm done

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": "eslint:recommended",
+    "parserOptions": {
+        "ecmaVersion": 12,
+        "sourceType": "module"
+    },
+    "rules": {
```
Comment (malaba6) [inline .eslintrc.json]: Can you plz define at least one rule here, for example the indent rule.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+app.use(async (req, res) => {
+    const error = await new Error('Not found');
+    error.status = 404;
+    res.send({status: error.status, message: error.message});
+});
+
+
+app.listen(port, console.log(`server has started on port ${port}`));
+
+export default app;
```
Comment (malaba6) [inline src/index.js]: anable eslint and errors like 'no new line at the end of the file' will go way

**Review by malaba6 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (Isaackomeza) [reply src/index.js]: It's Okay now

**Comment by Isaackomeza** [comment]
[Set up Node.js REST API empty project.](https://trello.com/c/qmeEmEod/43-set-up-nodejs-rest-api-empty-project)

**Review by malaba6 — APPROVED** [review]

**Pull request merged**